library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -517,6 +552,18 @@ public void teardown()
   }
 
   @Override
+  public void activate(OperatorContext context)
+  {
+    scanDirService.scheduleAtFixedRate(new DirectoryScannerThread(), 0, scanIntervalMillis, TimeUnit.MILLISECONDS);
```

Comment:
@amberarrow Currently the scanDirService is defined as  Executors.newScheduledThreadPool(1), which returns type ScheduledThreadPoolExecutor. I've updated the method to scheduleWithFixedDelay() to avoid building the queue.